Solutions/CohesitySecurity/Tools/deploy_playbooks.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/zsh
+set -e
+
+# Description: This script deploys Azure Sentinel playbooks for Cohesity
+
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+# Validate input variables
+if [[ -z "$resource_group" ]]; then
+    echo "Error: Resource Group is not set. Please set the 'resource_group' variable."
+    exit 1
+fi
+
+# Function to deploy the playbook
+deploy_playbook() {
+    playbook_name="$1"
+    template_file="../Playbooks/$playbook_name/azuredeploy.json"
+
+    if [ -f "$template_file" ]; then
+        echo "Deploying $playbook_name..."
+        az deployment group create \
+            --name "${playbook_name}_Deployment" \
+            --resource-group "$resource_group" \
+            --template-file "$template_file" \
+            --parameters PlaybookName="$playbook_name"
+        echo "$playbook_name deployment complete."
+    else
+        echo "Error: Template file for $playbook_name not found."
+    fi
+}
+
+# Deploy the playbooks
+playbook_names=("Cohesity_Close_Helios_Incident" "Cohesity_CreateOrUpdate_ServiceNow_Incident" "Cohesity_Delete_Incident_Blobs" "Cohesity_Restore_From_Last_Snapshot")
```

Comment:
Your inquiry is well taken. The playbook `Cohesity_Send_Incident_Email` necessitates additional parameters for deployment. Thus, it necessitates its removal from the current list, requiring a unique deployment script.

Similarly, the playbook `Cohesity_CreateOrUpdate_ServiceNow_Incident` is also presumed to need a distinct configuration script. Consequently, the proposed change to unify all playbook deployments into a single script for code reusability is, unfortunately, impractical.

Additionally, the current code configuration makes debugging quite challenging:
```
#!/bin/zsh
set -e
```
Your proposed modification alters the error message from 
```
"Failed to deploy the Cohesity_Send_Incident_Email."
```
to 
```
ERROR: Missing input parameters: EmailID
```
The latter message lacks specificity and fails to clearly indicate the source of the error. This lack of clarity makes debugging even more complex.

Reply:
My proposed modification was a suggestion. It would have been nice to test my proposal and revert back with the issue. I assumed that line number will be printed at the point of the failure. 

Btw, it's very simple to fix it by doing something like [this](https://unix.stackexchange.com/questions/39623/trap-err-and-echoing-the-error-line).